lib/health_check.rb

```diff
@@ -0,0 +1,91 @@
+class HealthCheck
+  include ChefResource
+
+  REACHABLE = 'reachable'.freeze
+  TIMEOUT = 'timeout'.freeze
+  UNREACHABLE = 'unreachable'.freeze
+  ERRORING = 'erroring'.freeze
+  KEYERROR = 'key error'.freeze
+
+  attr_reader :status, :erchef, :postgres
+
+  def initialize
+    @status = nil
+    @erchef = { status: REACHABLE }
+    @postgres = { status: REACHABLE }
+  end
+
+  #
+  # Do a general health check.
+  #
+  def check
+    erchef_health
+    erchef_authentication
+    postgres_health
+    overall
+  end
+
+  private
+
+  #
+  # Try to ping erchef as an indicator of general health
+  #
+  def erchef_health
+    erchef_health_metric do
+      chef.get_rest 'license'
```

Comment:
We should also ask on any thoughts around how to validate the authentication and if that is even something we really need to pursue or if we're trying to do more than we should.